shared/sliceutil/slice.go

```diff
@@ -302,3 +304,62 @@ func SplitCommaSeparated(arr []string) []string {
 func SplitOffset(listSize, chunks, index uint64) uint64 {
 	return (listSize * index) / chunks
 }
+
+// IntersectionSlot of any number of types.Slot slices with time
+// complexity of approximately O(n) leveraging a map to
+// check for element existence off by a constant factor
+// of underlying map efficiency.
+func IntersectionSlot(s ...[]types.Slot) []types.Slot {
```

Comment:
Are there tests for this function?

Reply:
Good catch, I will include the tests for all three methods in this file.